server/src/handle_chaos_message.rs

```diff
@@ -453,85 +497,87 @@ fn make_messages(proto_messages: &[chaos_proto::chaos::Message]) -> Result<Vec<M
+                let channel_type = channel_type
+                    .enum_value()
+                    .map_err(|value| format_err!("'{}' is not a valid 'channel::Type'", value))?;
+                let channel_type = make_channel_type(channel_type);
+                Ok::<_, Error>(channel_type)
+            })
+            .collect::<Result<_, _>>()?,
     };
     Ok(result)
 }
 
-fn make_channel_type(proto: chaos_proto::chaos::Channel_Type) -> ChannelType {
-    use chaos_proto::chaos::Channel_Type;
+fn make_channel_type(proto: chaos_proto::chaos::channel::Type) -> ChannelType {
+    use chaos_proto::chaos::channel;
     match proto {
-        Channel_Type::web => ChannelType::Web,
-        Channel_Type::sms => ChannelType::Sms,
-        Channel_Type::email => ChannelType::Email,
-        Channel_Type::mobile => ChannelType::Mobile,
-        Channel_Type::notification => ChannelType::Notification,
-        Channel_Type::twitter => ChannelType::Twitter,
-        Channel_Type::facebook => ChannelType::Facebook,
-        Channel_Type::unkown_type => ChannelType::UnknownType,
-        Channel_Type::title => ChannelType::Title,
-        Channel_Type::beacon => ChannelType::Beacon,
+        channel::Type::web => ChannelType::Web,
+        channel::Type::sms => ChannelType::Sms,
+        channel::Type::email => ChannelType::Email,
+        channel::Type::mobile => ChannelType::Mobile,
+        channel::Type::notification => ChannelType::Notification,
+        channel::Type::twitter => ChannelType::Twitter,
+        channel::Type::facebook => ChannelType::Facebook,
+        channel::Type::unkown_type => ChannelType::UnknownType,
+        channel::Type::title => ChannelType::Title,
+        channel::Type::beacon => ChannelType::Beacon,
     }
 }
 
 fn make_application_pattern(
     proto: &chaos_proto::chaos::Pattern,
 ) -> Result<ApplicationPattern, Error> {
-    let begin_date = if proto.has_start_date() {
-        let timestamp = proto.get_start_date();
-        let datetime = NaiveDateTime::from_timestamp(i64::from(timestamp), 0);
-        datetime.date()
-    } else {
-        bail!("Pattern has no start_date");
-    };
-    let end_date = if proto.has_end_date() {
-        let timestamp = proto.get_end_date();
-        let datetime = NaiveDateTime::from_timestamp(i64::from(timestamp), 0);
-        datetime.date()
-    } else {
-        bail!("Pattern has no end_date");
-    };
-    let time_slots = proto.get_time_slots().iter().map(make_timeslot).collect();
-    if proto.has_end_date() {
-        let timestamp = proto.get_end_date();
-        let datetime = NaiveDateTime::from_timestamp(i64::from(timestamp), 0);
-        datetime.date()
-    } else {
-        bail!("Pattern has no end_date");
-    };
+    let begin_date = proto
+        .start_date
+        .ok_or_else(|| format_err!("'Pattern' has no 'start_date'"))?;
+    let begin_date = make_datetime(begin_date as u64)?.date(); // u32 can always be coerced to u64
```

Comment:
```suggestion
    let begin_date = make_datetime(u64::from(begin_date))?.date(); 
```
spotted by [SonarCloud](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=224&id=Hove_loki&open=AYDh_xF9TOKYc_BU_D7C) :+1:

Reply:
Done in 02a12ac.